### ISSUE TITLE
problem in using ddp

### ISSUE THREAD

**Issue by toyottttttt** [open-mmlab/mmdetection#10234]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

Reproduces the problem - code sample

https://github.com/sean-zhuh/SeqTR/

Reproduces the problem - command or script

train:

bash tools/dist_train.sh configs/seqtr/detection/seqtr_det_refcoco-unc.py --cfg-options ema=True

test:
python tools/test.py configs/seqtr/detection/seqtr_det_refcoco-unc.py --load-from [PATH_TO_CHECKPOINT_FILE] --cfg-options ema=True

Reproduces the problem - error message

When running the code, a validation set is tested every epoch, and the best result achieved is around 80%. However, when using the saved test set for testing, the result is only around 30%.

(1) This issue does not occur when using single GPU. When training and testing with a single GPU, the results are correct regardless of whether it is a single or multiple gpu test.

(2) Some one has also encountered the same issue in their issue tracker(https://github.com/sean-zhuh/SeqTR/issues/4). The author responded that there may be bugs when using multiple GPUs, which may be caused by the mmcv multiple GPU training module. We have spent a lot of time debugging but have not been able to identify where the issue lies.

**Environment**

install:
torch: 1.9.0+cu111;
torchvision: 0.10.0+cu111
mmcv-full:1..5.0
mmdet: 2.21.0

pip install torch==1.9.0+cu111 torchvision==0.10.0+cu111 torchaudio==0.9.0 -f https://download.pytorch.org/whl/torch_stable.html
pip install mmdet==2.21.0
pip install mmcv-full=1.5.0 -f https://download.openmmlab.com/mmcv/dist/cu111/torch1.9.0/index.html
